app/src/main/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntryEffectHandler.kt

```diff
@@ -146,14 +145,10 @@ class BloodSugarEntryEffectHandler @AssistedInject constructor(
   }
 
   private fun userAndCurrentFacility(): Single<Pair<User, Facility>> {
-    return userSession
-        .loggedInUser()
-        .filterAndUnwrapJust()
-        .flatMap { user ->
-          facilityRepository
-              .currentFacility(user)
-              .map { facility -> user to facility }
-        }
+    val user = currentUser.get()
+    return facilityRepository
+        .currentFacility(user)
+        .map { facility -> user to facility }
```

Comment:
We can also inject the the current `Facility` directly, right?

Reply:
Yes we can. I didn't do this because I was only injecting `user` at first. Also this required changing the streams that were consuming this. I will make this change.